# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1202400415 -3600
#      Thu Feb 07 17:06:55 2008 +0100
# Node ID f79dcba526a840aac7699e9ba8fe55f8bae5eabc
# Parent  cb4838d70ab23a31999d0971f5a10f167ea2eced
Export nested properties classes of all graphics object.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,16 @@
 2008-06-04  Michael Goffioul <michael.goffioul@gmail.com>
 
+	* graphics.h.in (class root_figure::properties,
+	class line::properties, class text::properties,
+	class image::properties, class patch::properties,
+	class surface::properties): Export classes with
+	OCTINTERP_API.
+
 	* graphics.cc (axes::properties::set_defaults): Use correct
 	default values for gridlinestyle and minorgridlinestyle.
 
 2008-06-03  Jaroslav Hajek <highegg@gmail.com>
 
 	* load-save.cc (Fload): Fix "-7" option.
 
 2008-06-02  David Bateman  <dbateman@free.fr>
diff --git a/src/graphics.h.in b/src/graphics.h.in
--- a/src/graphics.h.in
+++ b/src/graphics.h.in
@@ -1981,17 +1981,17 @@ private:
   base_graphics_object *rep;
 };
 
 // ---------------------------------------------------------------------
 
 class OCTINTERP_API root_figure : public base_graphics_object
 {
 public:
-  class properties : public base_properties
+  class OCTINTERP_API properties : public base_properties
   {
   public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
 
     BEGIN_PROPERTIES(root_figure)
       handle_property currentfigure S , graphics_handle ()
     END_PROPERTIES
@@ -2626,17 +2626,17 @@ private:
   property_list default_properties;
 };
 
 // ---------------------------------------------------------------------
 
 class OCTINTERP_API line : public base_graphics_object
 {
 public:
-  class properties : public base_properties
+  class OCTINTERP_API properties : public base_properties
   {
   public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
 
     // properties which are not in matlab:
     // ldata, udata, xldata, xudata, keylabel, interpreter
 
@@ -2681,17 +2681,17 @@ public:
   bool valid_object (void) const { return true; }
 };
 
 // ---------------------------------------------------------------------
 
 class OCTINTERP_API text : public base_graphics_object
 {
 public:
-  class properties : public base_properties
+  class OCTINTERP_API properties : public base_properties
   {
   public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
 
     BEGIN_PROPERTIES(text)
       string_property string , ""
       radio_property units , "{data}|pixels|normalized|inches|centimeters|points"
@@ -2742,17 +2742,17 @@ public:
   bool valid_object (void) const { return true; }
 };
 
 // ---------------------------------------------------------------------
 
 class OCTINTERP_API image : public base_graphics_object
 {
 public:
-  class properties : public base_properties
+  class OCTINTERP_API properties : public base_properties
   {
   public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
 
     BEGIN_PROPERTIES(image)
       data_property xdata l , Matrix ()
       data_property ydata l , Matrix ()
@@ -2785,17 +2785,17 @@ public:
   bool valid_object (void) const { return true; }
 };
 
 // ---------------------------------------------------------------------
 
 class OCTINTERP_API patch : public base_graphics_object
 {
 public:
-  class properties : public base_properties
+  class OCTINTERP_API properties : public base_properties
   {
   public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
 
     BEGIN_PROPERTIES(patch)
       data_property xdata l , Matrix ()
       data_property ydata l , Matrix ()
@@ -2858,17 +2858,17 @@ public:
   bool valid_object (void) const { return true; }
 };
 
 // ---------------------------------------------------------------------
 
 class OCTINTERP_API surface : public base_graphics_object
 {
 public:
-  class properties : public base_properties
+  class OCTINTERP_API properties : public base_properties
   {
   public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
 
     BEGIN_PROPERTIES(surface)
       data_property xdata l , Matrix ()
       data_property ydata l , Matrix ()
